tla-types/src/main/scala/at/forsyte/apalache/tla/typecmp/subbuilder/SetBuilder.scala

```diff
@@ -0,0 +1,113 @@
+package at.forsyte.apalache.tla.typecmp.subbuilder
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper.TlaOper
+import at.forsyte.apalache.tla.typecmp.BuilderUtil._
+import at.forsyte.apalache.tla.typecmp.{BuilderWrapper, InternalState, NameWrapper, ValWrapper}
+import at.forsyte.apalache.tla.typecmp.raw.RawSetBuilder
+import scalaz._
+import scalaz.Scalaz._
+
+/**
+ * Builder for TlaSetOper expressions.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+trait SetBuilder extends RawSetBuilder {
+
+  /** {x1, ..., xn}, must have >0 args. */
+  def enumSet(argW: BuilderWrapper, argsW: BuilderWrapper*): BuilderWrapper = for {
```

Comment:
To me, `require` is a necessary evil in methods which both a) take a variable number of arguments, and b) impose constraints only on _some_ of the arguments (e.g. every other arg. has to be a `NameEx` or a string `ValEx`).
In your case, you can do 
```scala
val h +: tail = args
enumSet(h, tail:_*)
```
instead of `enumSet(args:_*)`, so it shouldn't be too bad.

Reply:
I won't block on this, but I would like to hear others' thoughts.

Reply:
Pinging @shonfeder @konnov @gabrielamafra

Reply:
Yeah, I also wrote above that I did not like that about the old builder. If we want to enforce some properties of a sequence at compile-time, we should not put this burden on the user. And @shonfeder mentioned shapeless.

Reply:
Ok, so the general sentiment seems to be to enforce minimal arity via runtime checks, so I'll go ahead and change those methods. 
Just one question, regarding maps (or methods with asymmetric args in general): do you prefer
`builder.map(e: T, xsAndSs: T*)` + `require(P(xsAndSs))` or `builder.map(eAndXsAndSs: T*)` + `require(P(eAndXsAndSs))`. Here, `e` and `xsAndSs` play different roles, and it's very unlikely that `eAndXsAndSs` would be programmatically constructed, since the components have different roles and shapes (unlike `xsAndSs`, which is uniform in shape).
I'm more in favor of the former, but I'd like to hear your opinions.

Reply:
I would definitely prefer `builder.map(e: T, xsAndSs: T*)`

Reply:
+1